lib/services/task_list_service.dart

```diff
@@ -26,6 +27,58 @@ class TaskListService with LogMixin, ChangeCallbackProvider {
     return (await _taskListCrdt).values;
   }
 
+  Future<List<Task>> getTasksForList(TaskList taskList) async {
+    var filtered = <Task>[];
+
+    List allTasks = await tasks;
+    for (var i = 0; i < allTasks.length; i++) {
+      if (allTasks[i].taskListID == taskList.id) {
+        filtered.add(allTasks[i]);
+      }
+    }
+
+    switch (taskList.sortBy) {
+      case SortOption.Title:
+        filtered
+            .sort((a, b) => a.title.toString().compareTo(b.title.toString()));
+        break;
+      case SortOption.Flag:
+        filtered.sort((a, b) {
+          if (b.isFlagged) {
+            return 1;
+          }
+
+          return -1;
+        });
+        break;
+      case SortOption.Status:
+        filtered.sort((a, b) {
+          if (b.completed) {
+            return -1;
+          }
+
+          return 1;
+        });
+        break;
+      case SortOption.DueDate:
+        filtered.sort((a, b) {
+          if (b.due == null) {
+            return -1;
+          }
+          if (a.due == null) {
+            return 1;
+          }
+
+          return a.due!.compareTo(b.due!);
+        });
+        break;
+      case SortOption.Created:
+        break;
```

Comment:
Since there only is a break, does the code assume that the tasks are always sorted by timestamp? Is this sorting guaranteed?